fastq_filtrator.py

```diff
@@ -0,0 +1,116 @@
+import argparse
+
+
+def passed_gc_filter(seq, threshold_dict):
+    threshold = threshold_dict["gc_bounds"]
+    gc_content = ((seq[1].count('G') + seq[1].count('C')) / len(seq[1])) * 100
+    if gc_content >= threshold[0] and gc_content <= threshold[1]:
+        return True
+    else:
+        return False
+
+
+def passed_quality_filter(seq_block, threshold_dict):
+    threshold = threshold_dict["quality_threshold"]
+    seq_score = []
+    for score in seq_block[3]:
+        seq_score.append(ord(score) - 33)
+    if len(seq_score) != 0:
+        avr_quality = sum(seq_score) / len(seq_score)
+        if avr_quality >= threshold:
+            return True
+        else:
+            return False
+    else:
+        return False
+
+
+def passed_length_filter(seq, threshold_dict):
+    threshold = threshold_dict["length_bounds"]
+    length = len(seq[1])
+    if length >= threshold[0] and length <= threshold[1]:
+        return True
+    else:
+        return False
+
+
+FILTERS = [passed_gc_filter,
+           passed_quality_filter,
+           passed_length_filter]
+
+
+def main(input_fastq, output_file_prefix, gc_bounds, length_bounds, quality_threshold, save_filtered):
```

Comment:
Аргументы в функции должны были иметь значения по-умолчанию. Я вижу, что дефолтные заданы при парсинге аргументов, но лучше сделать double check и указать в обоих местах.
```suggestion
def main(input_fastq, output_file_prefix, gc_bounds=(0, 100), length_bounds=(0, 2**32), quality_threshold=0, save_filtered=True):
```